core/src/main/scala/akka/kafka/scaladsl/Consumer.scala

```diff
@@ -55,6 +67,43 @@ object Consumer {
     def metrics: Future[Map[MetricName, Metric]]
   }
 
+  /**
+   * Combine control and a stream completion signal materialized values into
+   * one, so that the stream can be stopped in a controlled way without losing
+   * commits.
+   */
+  final class DrainingControl[T] private (tuple: (Control, Future[T])) extends Control {
```

Comment:
We could have the private constructor accept two parameters. And the deconstruction from a tuple to two parameters then can be moved to the factory methods.